openedx/openedx-platform
# feat: Add scope user_id to JWT payload

## Description

Ecommerce system uses the JWT payload attribute `user_id` to populate the field `lms_user_id` in the Ecommerce User table.
The JWT payload is built on edx-platform and sent to the Ecommerce system. Currently the attribute user_id is missing from this payload. This PR adds the missing attribute user_id to the JWT payload.

## Supporting information

Jira ticket: https://2u-internal.atlassian.net/browse/LEARNER-9640
Related ADR: https://github.com/openedx/ecommerce/blob/master/docs/decisions/0004-unique-identifier-for-users.rst
Related Ecommerce system attribute mapping: https://github.com/openedx/ecommerce/blob/f0e196fe4371f56a14969a1fb0d2fed79c39630a/ecommerce/settings/base.py#L654

## Testing instructions

1. Login a user via JWT
2. Use the JWT token to call an Ecommerce API for e.g. `{{ecommerce_domain}}/api/iap/v1/basket/add/?sku={{sku}} GET`
3. Place a debugger in the edx-drf-extensions in the ecommerce system in https://github.com/openedx/edx-drf-extensions/blob/e700fbffedde78963ae1e46d4267d10787dd9d0e/edx_rest_framework_extensions/auth/jwt/authentication.py#L129 and ensure the `payload` has the `user_id` included in it.

## Deadline

None
